src/combinatorics/quiver/path.lean

```diff
@@ -59,6 +65,36 @@ def comp {a b : V} : Π {c}, path a b → path b c → path a c
 | c p q path.nil := rfl
 | d p q (path.cons r e) := by rw [comp_cons, comp_cons, comp_cons, comp_assoc]
 
+/-- Turn a path into a list. The list contains `a` at its head, but not `b` a priori. -/
+@[simp] def to_list : Π {b : V}, path a b → list V
+| b nil := []
+| b (@cons _ _ _ c _ p f) := c :: p.to_list
+
+@[simp] lemma to_list_comp (p : path a b) :
+  ∀ {c} (q : path b c), (p.comp q).to_list = q.to_list ++ p.to_list
+| c nil := by simp
+| c (@cons _ _ _ d _ q f) := by simp [to_list_comp]
+
+lemma to_list_chain_nonempty :
+  ∀ {b} (p : path a b), p.to_list.chain (λ x y, nonempty (y ⟶ x)) b
```

Comment:
I was expecting this to read
```suggestion
  ∀ {b} (p : path a b), p.to_list.chain (λ x y, nonempty (x ⟶ y)) b
``
Am I confused about the direction of `chain`, `path`, or `⟶`? A comment might be worth having on either this lemma or `to_list` to make this convention clear.

Reply:
Yes indeed the order is reversed between `list` and `quiver`. This is why ``to_list_comp` reads backwards. I believe this is a conscious design decision.